tfhe/benches/core_crypto/pbs_bench.rs

```diff
@@ -547,3 +536,290 @@ fn pbs_throughput<Scalar: UnsignedTorus + CastInto<usize> + Sync + Send + Serial
+                params.lwe_dimension.unwrap(),
+                &mut secret_generator,
+            );
+            let output_glwe_secret_key: GlweSecretKeyOwned<Scalar> =
+                allocate_and_generate_new_binary_glwe_secret_key(
+                    params.glwe_dimension.unwrap(),
+                    params.polynomial_size.unwrap(),
+                    &mut secret_generator,
+                );
+            let output_lwe_secret_key = output_glwe_secret_key.into_lwe_secret_key();
+
+            let bsk = LweBootstrapKey::new(
+                Scalar::ZERO,
+                params.glwe_dimension.unwrap().to_glwe_size(),
+                params.polynomial_size.unwrap(),
+                params.pbs_base_log.unwrap(),
+                params.pbs_level.unwrap(),
+                params.lwe_dimension.unwrap(),
+                tfhe::core_crypto::prelude::CiphertextModulus::new_native(),
+            );
+            let bsk_gpu = CudaLweBootstrapKey::from_lwe_bootstrap_key(&bsk, &stream);
+
+            // Allocate a new LweCiphertext and encrypt our plaintext
+            let lwe_ciphertext_in = allocate_and_encrypt_new_lwe_ciphertext(
+                &input_lwe_secret_key,
+                Plaintext(Scalar::ZERO),
+                params.lwe_modular_std_dev.unwrap(),
+                tfhe::core_crypto::prelude::CiphertextModulus::new_native(),
+                &mut encryption_generator,
+            );
+            let lwe_ciphertext_in_gpu =
+                CudaLweCiphertextList::from_lwe_ciphertext(&lwe_ciphertext_in, &stream);
+
+            let accumulator = GlweCiphertext::new(
+                Scalar::ZERO,
+                params.glwe_dimension.unwrap().to_glwe_size(),
+                params.polynomial_size.unwrap(),
+                tfhe::core_crypto::prelude::CiphertextModulus::new_native(),
+            );
+            let accumulator_gpu =
+                CudaGlweCiphertextList::from_glwe_ciphertext(&accumulator, &stream);
+
+            // Allocate the LweCiphertext to store the result of the PBS
+            let mut out_pbs_ct = LweCiphertext::new(
+                Scalar::ZERO,
+                output_lwe_secret_key.lwe_dimension().to_lwe_size(),
+                tfhe::core_crypto::prelude::CiphertextModulus::new_native(),
+            );
+            let mut out_pbs_ct_gpu =
+                CudaLweCiphertextList::from_lwe_ciphertext(&out_pbs_ct, &stream);
+            let h_indexes = &[Scalar::ZERO];
+            stream.synchronize();
+            let mut d_input_indexes = stream.malloc_async::<Scalar>(1u32);
+            let mut d_output_indexes = stream.malloc_async::<Scalar>(1u32);
+            let mut d_lut_indexes = stream.malloc_async::<Scalar>(1u32);
+            stream.copy_to_gpu_async(&mut d_input_indexes, h_indexes.as_ref());
+            stream.copy_to_gpu_async(&mut d_output_indexes, h_indexes.as_ref());
+            stream.copy_to_gpu_async(&mut d_input_indexes, h_indexes.as_ref());
```

Comment:
I think you'll have to rebase and put that in a unsafe block due to https://github.com/zama-ai/tfhe-rs/commit/11db96d3941328cd18f0d04e9dd68a512d4bef07 being merged